execution/block-partitioner/src/sharded_block_partitioner/conflict_detector.rs

```diff
@@ -72,9 +74,8 @@ impl CrossShardConflictDetector {
         current_round_rw_set_with_index: Arc<Vec<WriteSetWithTxnIndex>>,
         prev_rounds_rw_set_with_index: Arc<Vec<WriteSetWithTxnIndex>>,
     ) -> CrossShardDependencies {
-        if current_round_rw_set_with_index.is_empty() && prev_rounds_rw_set_with_index.is_empty() {
```

Comment:
Why remove this? This makes it efficient in first round to early return.

Reply:
`current_round_rw_set_with_index` won't be empty with the proposed fix (see [this](https://github.com/aptos-labs/aptos-core/pull/8826#discussion_r1241302932) thread), making it unreachable.